libraries/core_libs/consensus/src/transaction_manager/transaction_manager.cpp

```diff
@@ -278,6 +278,20 @@ void TransactionManager::updateFinalizedTransactionsStatus(SyncBlock const &sync
   db_->saveStatusField(StatusDbField::TrxCount, trx_count_);
 }
 
+void TransactionManager::moveNonFinalizedTransactionsToTransactionsPool(std::unordered_set<trx_hash_t> &&transactions) {
+  // !!! There is no lock because it is called under std::unique_lock trx_lock(trx_mgr_->getTransactionsMutex());
+  auto write_batch = db_->createWriteBatch();
+  for (auto const &trx_hash : transactions) {
+    db_->removeTransactionToBatch(trx_hash, write_batch);
```

Comment:
should be this inside `if (trx != nonfinalized_transactions_in_dag_.end()) {` ?

Reply:
Yes, good catch. Fixed